frappe/model/base_document.py

```diff
@@ -70,6 +70,25 @@
 )
 
 
+def _reconstruct_extended_instance(doctype, state):
```

Comment:
this is faster than regular pickling / unpickling for some reason. perhaps we can do this for all documents in a separate PR.